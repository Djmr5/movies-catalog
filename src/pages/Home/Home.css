.section-header {
    @apply flex justify-between items-center pr-2;
}

.section-title {
    @apply text-3xl font-medium uppercase pl-10 py-5;
}

.movie-list {
    @apply flex overflow-x-auto;
    /* Hide scrollbar */
    -ms-overflow-style: none; /* for Internet Explorer, Edge */
    scrollbar-width: none; /* for Firefox */
}

.movie-list::-webkit-scrollbar {
    /* Hide scrollbar */
    display: none; /* for Chrome, Safari, and Opera */
}
