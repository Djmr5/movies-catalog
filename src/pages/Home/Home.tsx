import { MovieCard, NavButton } from "../../components";
import { movies } from "../../constants/moviesMock"
import './Home.css';

const Home = () => {
  return (
    <>
      <section id="popular-section">
        <div className="section-header">
          <h3 className="section-title">Popular</h3>
          <NavButton text={"View All"} color={"red"} link={"popular"} />
        </div>
        <div className="movie-list">
          <MovieCard
            movieId={movies[0].id}
            title={movies[0].title}
            genreId={movies[0].genre_ids[0]}
            voteAverage={movies[0].vote_average}
            posterPath={movies[0].poster_path}
          />
          <MovieCard
            movieId={movies[1].id}
            title={movies[1].title}
            genreId={movies[1].genre_ids[1]}
            voteAverage={movies[1].vote_average}
            posterPath={movies[1].poster_path}
          />
          <MovieCard
            movieId={movies[2].id}
            title={movies[2].title}
            genreId={movies[2].genre_ids[2]}
            voteAverage={movies[2].vote_average}
            posterPath={movies[2].poster_path}
          />
          <MovieCard
            movieId={movies[3].id}
            title={movies[3].title}
            genreId={movies[3].genre_ids[3]}
            voteAverage={movies[3].vote_average}
            posterPath={movies[3].poster_path}
          />
          <MovieCard
            movieId={movies[4].id}
            title={movies[4].title}
            genreId={movies[4].genre_ids[4]}
            voteAverage={movies[4].vote_average}
            posterPath={movies[4].poster_path}
          />
          <MovieCard
            movieId={movies[5].id}
            title={movies[5].title}
            genreId={movies[5].genre_ids[5]}
            voteAverage={movies[5].vote_average}
            posterPath={movies[5].poster_path}
          />
          <MovieCard
            movieId={movies[6].id}
            title={movies[6].title}
            genreId={movies[6].genre_ids[6]}
            voteAverage={movies[6].vote_average}
            posterPath={movies[6].poster_path}
          />
        </div>
      </section>
      <MovieCard
        movieId={movies[0].id}
        title={movies[0].title}
        genreId={movies[0].genre_ids[0]}
        voteAverage={movies[0].vote_average}
        posterPath={movies[0].poster_path}
      />
    </>
  );
};

export default Home;
