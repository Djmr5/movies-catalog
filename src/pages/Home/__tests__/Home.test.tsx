import Home from "../Home";
import { expect, describe, it } from "vitest";
import { render } from "@testing-library/react";

describe("Tests for Home Page", () => {
  it("renders correctly", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it("tests", () => {
    expect(1).toBe(1);
  });
});
