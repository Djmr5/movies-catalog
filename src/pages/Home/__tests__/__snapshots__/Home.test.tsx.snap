// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for Home Page > renders correctly 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
